keanu-examples/bayesianMethodsForHackers/src/main/java/examples/TextMessaging.java

```diff
@@ -44,28 +47,31 @@ public static TextMessagingResults run() {
         PoissonVertex textsForDay = new PoissonVertex(rateForDay);
         textsForDay.observe(data.numberOfMessages);
 
-        BayesianNetwork net = new BayesianNetwork(switchPoint.getConnectedGraph());
+        BayesianNetwork net = new BayesianNetwork(textsForDay.getConnectedGraph());
+        net.probeForNonZeroProbability(1000);
 
-        int numSamples = 50000;
+        final int numSamples = 50000;
         NetworkSamples posteriorSamples = MetropolisHastings.withDefaultConfig()
             .getPosteriorSamples(net, net.getLatentVertices(), numSamples)
             .drop(numSamples / 10)
             .downSample(net.getLatentVertices().size());
 
-        int mostProbableSwitchPoint = posteriorSamples.getIntegerTensorSamples(switchPoint).getScalarMode();
-
-        WriteCsv.asSamples(posteriorSamples, switchPoint)
-            .withHeader("switchpoint")
-            .toFile("switchpoitsamples.csv");
-
-        return new TextMessagingResults(mostProbableSwitchPoint);
+        return new TextMessagingResults(
+            posteriorSamples.getIntegerTensorSamples(switchPoint).getScalarMode(),
+            posteriorSamples.getDoubleTensorSamples(earlyRate).getMode().scalar(),
```

Comment:
Getting the mode of a double might give you some strange behaviour without a bin size. Does this work when using getAverages instead?